src/main/java/org/apache/commons/lang3/builder/EqualsBuilder.java

```diff
@@ -0,0 +1,1145 @@
+     * (and non primitive wrapper) field types will be compared by
+     * {@code EqualsBuilder} recursively instead of invoking their
+     * {@code equals()} method. Leading to a deep reflection equals test.
+     *
+     * @param lhs  {@code this} object
+     * @param rhs  the other object
+     * @param testTransients  whether to include transient fields
+     * @param reflectUpToClass  the superclass to reflect up to (inclusive),
+     *  may be {@code null}
+     * @param testRecursive  whether to call reflection equals on non primitive
+     *  fields recursively.
+     * @param excludeFields  array of field names to exclude from testing
+     * @return {@code true} if the two Objects have tested equals.
+     *
+     * @see EqualsExclude
+     * @since 3.6
+     */
+    public static boolean reflectionEquals(final Object lhs, final Object rhs, final boolean testTransients, final Class<?> reflectUpToClass,
+            final boolean testRecursive, final String... excludeFields) {
+        if (lhs == rhs) {
+            return true;
+        }
+        if (lhs == null || rhs == null) {
+            return false;
+        }
+        return new EqualsBuilder()
+                    .setExcludeFields(excludeFields)
+                    .setReflectUpToClass(reflectUpToClass)
+                    .setTestTransients(testTransients)
+                    .setTestRecursive(testRecursive)
+                    .reflectionAppend(lhs, rhs)
+                    .isEquals();
+    }
+
+    /**
+     * <p>Tests if two {@code objects} by using reflection.</p>
+     *
+     * <p>It uses {@code AccessibleObject.setAccessible} to gain access to private
+     * fields. This means that it will throw a security exception if run under
+     * a security manager, if the permissions are not set up correctly. It is also
+     * not as efficient as testing explicitly. Non-primitive fields are compared using
+     * {@code equals()}.</p>
+     *
+     * <p>If the testTransients field is set to {@code true}, transient
+     * members will be tested, otherwise they are ignored, as they are likely
+     * derived fields, and not part of the value of the {@code Object}.</p>
+     *
+     * <p>Static fields will not be included. Superclass fields will be appended
+     * up to and including the specified superclass in field {@code reflectUpToClass}.
+     * A null superclass is treated as java.lang.Object.</p>
+     *
+     * <p>Field names listed in field {@code excludeFields} will be ignored.</p>
+     *
+     * <p>If either class of the compared objects is contained in
+     * {@code bypassReflectionClasses}, both objects are compared by calling
+     * the equals method of the left hand object with the right hand object as an argument.</p>
+     *
+     * @param lhs  the left hand object
+     * @param rhs  the left hand object
+     * @return EqualsBuilder - used to chain calls.
+     */
+    public EqualsBuilder reflectionAppend(final Object lhs, final Object rhs) {
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

The cyclomatic complexity of this method is 16. By comparison, 99% of the methods in the CodeGuru reference dataset have a lower cyclomatic complexity. This indicates the method has a high number of decisions and it can make the logic difficult to understand and test. We recommend that you simplify this method or break it into multiple methods.